src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -582,21 +584,24 @@ class OpenShiftDSL implements Serializable {
             obj = JsonOutput.toJson( template );
         }
 
-        String target = obj.toString();
-        if ( target.contains("{") ) { // is the string JSON?
-            String json = obj.toString();
-            FilePath f = currentContext.exec.getWorkspaceFilePath().createTextTempFile( "process", ".json", json, false );
+        String s = obj.toString();
+        boolean markup = s.contains("{") || s.contains(":")
+        boolean httpref = s.toLowerCase().startsWith("http")
+        Result r = new Result( "process" )
+        if ( markup && !httpref ) { // does this look like json or yaml?
+            FilePath f = currentContext.exec.getWorkspaceFilePath().createTextTempFile( "process", ".markup", s, false );
             try {
-                r = new Result( "process" );
                 r.actions.add((OcAction.OcActionResult)script._OcAction( buildCommonArgs("process", ["-f", f.getRemote() ], args, "-o=json" ) ));
                 r.failIf( "process returned an error" );
             } finally {
                 f.delete();
             }
+        } else if (httpref) {
```

Comment:
if it starts with a { why would we ever treat it as a http url?

Reply:
ok i guess i see how the logic works.  looks fragile but i guess this isn't making it worse.

Reply:
I tried to duplicate what was there, i plan on doing some further refactoring but didn't want to mix that in this pr